broken-links-script/cypress/e2e/link-checker.cy.js

```diff
@@ -117,12 +150,23 @@ describe('Link and Routing Validation - Individual URL Checks', () => {
         checkGithubFragment(fragment);
       }
       else if (fragment) {
-        cy.visit(url, { timeout: 20000 });
+        cy.visit(url, { timeout: 30000, failOnStatusCode: false, headers: { 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) Chrome/122.0 Safari/537.36' }});
         checkRegularFragment(fragment);
       }
       else {
-        cy.visit(url, { timeout: 20000 });
-        cy.document().its('body').should('not.be.empty');
+        cy.request({
+          url: url,
+          failOnStatusCode: false
+        }).then((response) => {
+          const contentType = response.headers['content-type'] || '';
+          if (!contentType.includes('text/html')) {
+            cy.log(`Non-HTML content detected for ${url}, skipping cy.visit()`);
+            expect(response.status).to.be.oneOf([200, 201, 202, 203, 204, 301, 302, 304]);
```

Comment:
Should we check that `response.body` is not empty? If we direct user there, there should be some content. Do you have any example of what non-html resource might that be?

Reply:
For example, Link: https://download.cumulocity.com/Apama/Debian/. These are file repositories or download links, not HTML pages, so I only check that they return a valid status and skip checking the body.

Reply:
This link seems to be special as it actually returns 2 `content-type` headers, but `response.headers` only shows you one of them.

<img width="1364" height="1065" alt="2025-10-29_17-35" src="https://github.com/user-attachments/assets/d8804d77-f0f4-45a8-82db-f3e9b72347c7" />

Anyway, I'd consider checking `response.body` to not be empty, just as you check detected `text/html` pages for non-empty HTML body.

```suggestion
            expect(response.status).to.be.oneOf([200, 201, 202, 203, 204, 301, 302, 304]);
            expect(response.body).not.to.be.empty;
```